### PR TITLE
Add IP search for addons & ratings admin

### LINKED ISSUE #20091: Display and search by ip address in ratings admin using IP address from Activity Log
Instead of relying on the `ip_address` field in the `Rating` model, we should use the `IPLog.ip_address_binary` field instead, both when searching and displaying.

While we're at it, allow searching by IP (which we record for each version submission, including the first) for add-ons.

Prerequisite for https://github.com/mozilla/addons-server/issues/19672

Depends on:
- https://github.com/mozilla/addons-server/issues/20135
- https://github.com/mozilla/addons-server/issues/20140

### PR DESCRIPTION
**Pull request by diox**
In addition:
- Store IPs for `EDIT_RATING` action as well (using it in search)
- Add `addon` filter to ratings, but only display available lookups (which can be very expensive to gather) if doing an IP search
- Make search fields more consistent between the different admins
- Add help text for search across all admins

Fixes #20091

### CODE REVIEW

File: src/olympia/abuse/tests/test_admin.py [hunk 1]
```diff
@@ -136,17 +136,7 @@ def test_search_user(self):
...
-        assert response.status_code == 200
-        doc = pq(response.content)
-        assert doc('#changelist-search')
-        assert doc('#result_list tbody tr').length == 1
-        assert 'Ehehehehe' in doc('#result_list').text()
-
-        user = AbuseReport.objects.get(message='Ehehehehe').user
-        response = self.client.get(
-            self.list_url, {'q': user.email[:3], 'type': 'user'}, follow=True
+            self.list_url, {'q': f'{user.email[:3]}*', 'type': 'user'}, follow=True
```
Comment (diox) [inline src/olympia/abuse/tests/test_admin.py:139]: Search by email in abuse reports is now a `LIKE` for consistency

File: src/olympia/addons/admin.py [hunk 2]
```diff
@@ -358,25 +365,27 @@ def reviewer_links(self, obj):
...
         if obj._listed_versions_exists:
             links.append(
-                '<a href="{}">{}</a>'.format(
+                (
                     urljoin(
                         settings.EXTERNAL_SITE_URL,
                         reverse('reviewers.review', args=['listed', obj.id]),
                     ),
-                    _('Reviewer Tools (listed)'),
+                    _('Review (listed)'),
```
Comment (diox) [inline src/olympia/addons/admin.py:373]: These changes are just to gain some horizontal space and be consistent, both in terms of copy and styling.

File: src/olympia/constants/activity.py [hunk 3]
```diff
@@ -466,6 +466,7 @@ class EDIT_RATING(_LOG):
     action_class = 'review'
     format = _('{rating} for {addon} updated.')
     show_user_to_developer = True
+    store_ip = True
```
Comment (eviljeff) [inline src/olympia/constants/activity.py:469]: oh! that seems an unrelated change - were we not storing the ip for rating edits before?  Is there something to backfill?

**Review by eviljeff — APPROVED** [review]

File: src/olympia/constants/activity.py (hunk 3 above)
Comment (diox) [reply src/olympia/constants/activity.py:469]: we were not. It's not possible to backfill this: we don't have the data

File: src/olympia/amo/admin.py [hunk 4]
```diff
@@ -41,6 +45,50 @@ class AMOModelAdminChangeList(ChangeList):
 
     search_form_class = AMOModelAdminChangeListSearchForm
 
+    def __init__(self, request, *args, **kwargs):
+        super().__init__(request, *args, **kwargs)
+        # django's ChangeList does:
+        # self.params = dict(request.GET.items())
+        # But we want to keep a QueryDict to not lose parameters present
+        # multiple times.
+        self.params = request.GET.copy()
```
[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/amo/admin.py:54]: because `QueryDict` can still be manipulated as a `dict`, it's fine. Note that we were already using this code in `ScannerResultAdmin`'s `ChangeList`, I just moved it to make it available to more admin tools that have filters that can be present multiple times.

File: src/olympia/abuse/admin.py [hunk 5]
```diff
@@ -237,11 +236,11 @@ def get_search_id_field(self, request):
...
         the type filter.
         """
-        type_ = request.GET.get('type')
+        type_ = request and request.GET.get('type')
         if type_ == 'user':
             search_field = 'user_id'
         else:
-            search_field = super().get_search_id_field(request)
+            search_field = None
         return search_field
```
[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/abuse/admin.py]: c9e71b0ebcb86b2a5fd85e3929c671cd7d34ad21
